src/soc/ibm/power9/i2c.c

```diff
@@ -176,6 +175,37 @@ int platform_i2c_transfer(unsigned int bus, struct i2c_msg *segment,
 
 	uint64_t clear_err = (type != FSI_I2C ? PPC_BIT(0) : 0);
 
+	/*
+	 * Divisor fields in this register are poorly documented:
+	 *
+	 *    Bits     SCOM   Field Mnemonic: Description
+	 *   0:7       RWX    BIT_RATE_DIVISOR_3: Decides the speed on the I2C bus.
+	 *   8:9       RWX    BIT_RATE_DIVISOR_3: Decides the speed on the I2C bus.
+	 *   10:15     RWX    BIT_RATE_DIVISOR_3: Decides the speed on the I2C bus.
+	 *
+	 * After issuing a fast command (SCOM A3000) they change like this:
+	 * -  100 kHz - previous value is not changed
+	 * -   50 kHz - 0x000B
+	 * - 3400 kHz - 0x005E
+	 * -  400 kHz - 0x0177
+	 *
+	 * Use value for 400 kHz as it is the one used by Hostboot.
+	 */
+	uint16_t bit_rate_div = 0x0177;	// 400kHz by default
+	if (bus == 2) {
+		/*
+		 * Skiboot computes the value as:
+		 *
+		 *   (((clock-frequency / bus-frequency) - 1) / 4)
```

Comment:
`clock-frequency` is frequency of core? If that is the case, it may cause problems for other buses, too. This also may mean that values change after OCC is started (boot frequency from scratch registers vs. target frequency from MVPD).

0x0177 was read on our CPU, but if it depends on CPU frequency, it may even be the cause of https://github.com/Dasharo/dasharo-issues/issues/80 if RCD is programmed incorrectly.

Reply:
It seems to come from skiboot. Clock frequency of i2cm is xscom frequency divided by 4. And for bus-frequency:
```c
		/*
		 * use the clock frequency as the bus frequency until we
		 * have actual devices on the bus. Adding a device will
		 * reduce the frequency to something that all devices
		 * can tolerate.
		 */
```
Although I don't see where it's updated.

Reply:
Ack. I expected it to come from machine XML but it doesn't seem to be the case.